geopandas/sindex.py

```diff
@@ -819,9 +819,12 @@ def intersection(self, coordinates):
         @property
         @doc(BaseSpatialIndex.size)
         def size(self):
-            return len(self)
+            return len(self._tree)
 
         @property
         @doc(BaseSpatialIndex.is_empty)
         def is_empty(self):
-            return len(self) == 0
+            return len(self._tree) == 0
+
+        def __len__(self):
+            return len(self._tree)
```

Comment:
This is the one thing I needed to add from `pygeos.STRtree` to get the sindex tests still passing. If we want to keep this in our sindex interface, I should maybe rather add it to the `BaseSpatialIndex` class. 

But I also don't know if this is necessarily needed, if there is the `.size` attribute (i.e. we could also deprecate being able to do `len(df.sindex)`)

Reply:
I have never really used the size/len attributes. After all, it's basically the same as the GeoDataFrame itself (minus some edge cases), and I'm not sure what you'd do with that information other than maybe compare sizes of two spatial indexes generated from many dataframes with empty geometries?

Reply:
I don't really mind either way. As @adriangb I have never had a need to use it in practice.

Reply:
I can see the need for us to use `len(self._tree) == 0` to test if underlying tree is empty (but that's also implementation specific to a tree engine), but I don't see clear value in providing that in the sindex API.  Seems like a good time to deprecate `__len__` here.